### PR TITLE
SCUMM: Add missing MI1 Lemonhead lines

### PR DESCRIPTION
**Pull request by eriktorbjorn**
Apparently some lines for Lemonhead went missing when The Secret From Monkey Island was updated for the CD version. It looks like a mistake, since one of the message that's still there isn't even displayed long enough to see. This pull request re-inserts the missing lines for the English CD version. Apparently localized versions are affected too, but I don't have them so all I can do is trying to make it as easy as possible to extend the workaround for them as well.

The "Ultimate Talkie" also has the bug. This pull request does nothing to fix that. I don't even know if the missing lines were recorded.

Unfortunately, my only savegame is based on boot params, not playing. (I did replay the game recently, but that was on a tablet and frog knows where those savefiles were hidden!)

[monkey1.s22.gz](https://github.com/scummvm/scummvm/files/6930299/monkey1.s22.gz)

### CODE REVIEW

File: engines/scumm/resource.cpp [hunk 1]
```diff
@@ -1771,5 +1781,115 @@ bool ScummEngine::verifyMI2MacBootScript(byte *buf, int size) {
...
+		// Insert the script patch and tag it with the appropriate
+		// language.
+
+		memcpy(scriptPtr + patchOffset, patchData, sizeof(patchData));
+		memcpy(scriptPtr + patchOffset + 7, lang, 3);
+
+		// Pad the rest of the replaced script part with spaces before
+		// terminating the string. Finally, add WaitForMessage().
+
+		memset(scriptPtr + patchOffset+ sizeof(patchData), 32, patchLength - sizeof(patchData) - 3);
```
Comment (mgerhardy) [inline engines/scumm/resource.cpp:1885]: a very very little minor tiny nitpick: the + is missing a space ;)

File: engines/scumm/script_v5.cpp [hunk 2]
```diff
@@ -2810,4 +2812,17 @@ void ScummEngine_v5::decodeParseString() {
...
 }
 
+void ScummEngine_v5::printPatchedMI1CannibalString(int textSlot, const byte *ptr) {
+	const char *msg = (const char *)ptr;
+
+	if (strncmp((const char *)ptr, "/LH.ENG/", 8) == 0) {
+		msg =
+"Oooh, that's nice.\xFF\x03"
+"Simple.  Just like one of mine.\xFF\x03"
+"And little.  Like mine.";
```
Comment (Mataniko) [inline engines/scumm/script_v5.cpp:2822]: Is the double space after the period in this line and the line above intentional?

Comment (eriktorbjorn) [reply engines/scumm/script_v5.cpp:2822]: > Is the double space after the period in this line and the line above intentional?

Yes, it's what I got from descumm with the floppy version's room-25-205s script:

```
[0086] (14) print(5,[Text("Take a look at this.")]);
[009E] (AE) WaitForMessage();
[00A0] (14) print(3,[Text("Oooh, that's nice." + wait() + "Simple.  Just like one of mine." + wait() + "And little.  Like mine.")]);
[00F0] (AE) WaitForMessage();
[00F2] (14) print(3,[Text("And it says, `Made by Lemonhead`^" + wait() + "^just like one of mine!" + wait() + "We should take this to the Great Monkey.")]);
[015A] (AE) WaitForMessage();
[015C] (14) print(5,[Text("Yes, I agree.")]);
[016D] (AE) WaitForMessage();
```

In the CD version, this was:

```
[0084] (14) print(5,[Text("Take a look at this.")]);
[009C] (AE) WaitForMessage();
[009E] (14) print(3,[Text("Oooh, that's nice.")]);
[00B4] (14) print(3,[Text("And it says, `Made by Lemonhead`^" + wait() + "^just like one of mine!" + wait() + "We should take this to the Great Monkey.")]);
[011C] (AE) WaitForMessage();
[011E] (14) print(5,[Text("Yes, I agree.")]);
[012F] (AE) WaitForMessage();
```

**Comment by JohnnyWalkerDesign** [comment]
Now fix the clocktower in Melee :)

**Comment by bluegr** [comment]
Nice work! +1 from me

**Comment by bluegr** [comment]
This is very nice work overall for the broken script, plus it adds the possibility of new script patches in the future. Thanks again for your work! Merging

**Pull request merged**